spec/src/main/asciidoc/batch_programming_model.adoc

```diff
@@ -1292,10 +1287,46 @@ The following example should make this more clear:
  // WONT WORK! - There is no property 'x' in scope for this injection
  @Inject @BatchProperty(name="x");
 
- // WILL WORK – Gets value 'xVal'
+ // WILL WORK - Gets value 'xVal'
  @Inject @BatchProperty(name="y");
 ----
 
+==== Unmatched property or empty property value does not necessarily get Java default value 
+
+If there is no matching JSL property for a given @BatchProperty name, or if the 
+corresponding JSL property has a value which resolves to the empty string (either 
+explicitly set to the empty string literal or resolving to an empty string via property
+substitution, as described in section xref:job-xml-substitution[8.8]), the resulting
+value is undefined by the batch specification. 
+
+It might be typical for a batch-managed instance to use the Java default value and
+a CDI implementation to set these @BatchProperty injection points to 'null', but
+users should not rely on consistent behaviors.   
+
+Instead @BatchProperty injection points in the Java code should correspond to matching 
+non-empty property values in the JSL. 
+
+==== Additional requirements for CDI bean instances
+
+===== Producer
```

Comment:
s/Producer/Bean/

a producer is a field or method with `@Producer` which enables the container to register a `Bean<?>` which is a generic available "factory". Consumers only see beans.

Do we want to support a few more primitive/wrappers? Any impl not doing it?

Lastly we must define what `@Dependent` gives, whicih is likely during component lookup (after thread local will be off in all impl) so it is not dependent on the job execution but on the initialization only, no?

Reply:
> a producer is a field or method with @Producer which enables the container to register a Bean<?> which is a generic available "factory". Consumers only see beans.

So we just say the batch runtime should ensure there is a Bean.. and whether it is done via Producer or some other method like an extension is just an implementation detail?    I think I got it.

> Do we want to support a few more primitive/wrappers? Any impl not doing it?

I was hoping to still do this in https://github.com/eclipse-ee4j/batch-api/issues/43  

>  Lastly we must define what @Dependent gives, whicih is likely during component lookup (after thread local will be off in all impl) so it is not dependent on the job execution but on the initialization only, no?

I was trying to say that the values would be set the same as in the **Properties for Dependent-scoped bean instances
** which implies we potentially have to wait until job execution for the JSL substitution to take place.    That's why I separated out Dependent scope from the other scopes.

Seems like we have a different take on this.    I'm thinking you do want to take advantage of JSL subsitution but you also want CDI injection, etc.   Do you see a use case where this isn't possible?